### ISSUE TITLE
vips-dev alpine - Error relocating /usr/lib/libheif.so.1: de265_get_image_full_range_flag: symbol not found

### ISSUE THREAD

**Issue by elrob** [lovell/sharp#2736]
In the past couple of days, something has changed on vips-dev (alpine) and its dependencies which means I'm getting the following error:
```
#10 0.665 Error:
#10 0.665 Something went wrong installing the "sharp" module
#10 0.665
#10 0.665 Error relocating /usr/lib/libheif.so.1: de265_get_image_full_range_flag: symbol not found
#10 0.665
#10 0.665 - Remove the "node_modules/sharp" directory then run
#10 0.665   "npm install --ignore-scripts=false --verbose sharp" and look for errors
#10 0.665 - Consult the installation documentation at https://sharp.pixelplumbing.com/install
#10 0.665 - Search for this error at https://github.com/lovell/sharp/issues
#10 0.665
#10 0.665     at Object.<anonymous> (/node_modules/sharp/lib/constructor.js:32:9)
#10 0.665     at Module._compile (internal/modules/cjs/loader.js:1063:30)
#10 0.665     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
#10 0.665     at Module.load (internal/modules/cjs/loader.js:928:32)
#10 0.665     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
#10 0.665     at Module.require (internal/modules/cjs/loader.js:952:19)
#10 0.665     at require (internal/modules/cjs/helpers.js:88:18)
#10 0.665     at Object.<anonymous> (/node_modules/sharp/lib/index.js:3:15)
#10 0.665     at Module._compile (internal/modules/cjs/loader.js:1063:30)
#10 0.665     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
```

The error can be reproduced with this dockerfile:

```
FROM node:14.16.1-alpine3.13 AS node-libvips-base
RUN apk --update add g++ make
ARG VIPS_VERSION=8.10.6-r1
RUN apk --update add -v vips-dev=$VIPS_VERSION --repository https://dl-cdn.alpinelinux.org/alpine/edge/community
RUN npm install sharp
RUN echo 'require("sharp")' > test.js
RUN node test.js
```

I think the significant difference between when it was working and now is the libheif(-dev) version:

```
#6 4.483 (109/137) Installing libheif (1.12.0-r0)
#6 4.522 (110/137) Installing libheif-dev (1.12.0-r0)
```

From CI logs, when this was `1.11.0-r0` rather than `1.12.0-r0` there was no issue.
However, it seems it's not possible to downgrade or pin that version with alpine.

1. Is it possible to resolve this issue?
2. Is the way I am attempting to use sharp and libvips part of the problem? Is there a recommended way to use sharp when I have the requirement for `heif` output?

**Comment by lovell** [comment]
Hi, it looks like you're installing packages from the Alpine edge repo into a non-edge environment, in this case Alpine 3.13.

>  Warning: "edge" is under constant development so be careful using it in production. It is possible that bugs in "edge" could cause data loss or could break your system.

https://wiki.alpinelinux.org/wiki/Edge

**Comment by elrob** [comment]
Thank you!
Resolved using non-edge vips-dev (on `8.10.5-r0`) and downgrading sharp to `0.27.2`.
```
FROM node:14.16.1-alpine3.13 AS node-libvips-base
RUN apk --update add g++ make vips-dev=8.10.5-r0
```
